.github/workflows/dev-build.yml

```diff
@@ -34,98 +33,59 @@ jobs:
-        with:
-          path: gradle.properties
-          property: release_version
-          value: ${{ steps.root_release.outputs.value }}
-      - name: Build custom grpc package version
-        id: python_grpc_ver
-        run: echo value="${{ steps.grpc_ver.outputs.release_version }}.dev${{ github.run_id }}" >> $GITHUB_OUTPUT
-      - name: Show app release version
-        run: echo ${{ steps.app_release_ver.outputs.value }}
-      - name: Show grpc java release version
-        run: echo ${{ steps.java_grpc_ver.outputs.value }}
-      - name: Show grpc python release version
-        run: echo ${{ steps.python_grpc_ver.outputs.value }}
-      - uses: actions/upload-artifact@v3
-        with:
-          name: java-versions
-          retention-days: 1
-          path: |
-            gradle.properties
-            **/gradle.properties
   build-and-publish-grpc-java:
     name: Build and publish Java gRPC distributions
     needs:
       - versions
-    uses: ./.github/workflows/build-and-publish-java.yml
+    uses: th2-net/.github/.github/workflows/compaund-java-multi-project-build.yml@multi-project-workflows
     with:
-      project-path: grpc
-      release: false
-      versions-artifact: java-versions
-    secrets: inherit
+      projectPath: grpc
+      additionalTasks: '["publish"]'
+      versionSuffix: ${{ needs.versions.outputs.version-suffix }}
+    secrets:
+      sonatypeUsername: ${{ secrets.SONATYPE_NEXUS_USERNAME }}
+      sonatypePassword: ${{ secrets.SONATYPE_NEXUS_PASSWORD }}
+      sonatypeSigningKey: ${{ secrets.SONATYPE_GPG_ARMORED_KEY }}
+      sonatypeSigningPassword: ${{ secrets.SONATYPE_SIGNING_PASSWORD }}
   build-and-publish-grpc-python:
     name: Build and publish Python distributions to PyPI
     needs:
       - versions
-    uses: ./.github/workflows/build-and-publish-python.yml
+    uses: th2-net/.github/.github/workflows/compaund-python-grpc-pypi-publication.yml@python
     with:
       custom-version: ${{ needs.versions.outputs.grpc-python }}
-    secrets: inherit
+      project-path: grpc
+    secrets:
+      pypi_password: ${{ secrets.PYPI_PASSWORD }}
   build-and-publish-core-java:
     name: Build and publish Java core distributions
     needs:
       - versions
-    uses: ./.github/workflows/build-and-publish-java.yml
+    uses: th2-net/.github/.github/workflows/compaund-java-multi-project-build.yml@multi-project-workflows
     with:
-      project-path: core
-      release: false
-      versions-artifact: java-versions
-    secrets: inherit
+      projectPath: core
+      additionalTasks: '["publish"]'
+      versionSuffix: ${{ needs.versions.outputs.version-suffix }}
+    secrets:
+      sonatypeUsername: ${{ secrets.SONATYPE_NEXUS_USERNAME }}
+      sonatypePassword: ${{ secrets.SONATYPE_NEXUS_PASSWORD }}
+      sonatypeSigningKey: ${{ secrets.SONATYPE_GPG_ARMORED_KEY }}
+      sonatypeSigningPassword: ${{ secrets.SONATYPE_SIGNING_PASSWORD }}
   owasp-scan:
```

Comment:
owasp-scan isn't required when we do trivy-docker-scan because it cover system, java vulnerabilities

Reply:
It doesn't look so. OWASP finds vulnerabilities that were not found by trivy

Reply:
Could you please share example.
We should choose one tool for testing or enable both in a specific case otherwise build will take a lot of time

Reply:
Here are the vulnerabilities found by OWASP
https://github.com/th2-net/th2-read-db/security/code-scanning?query=is%3Aopen+branch%3Aextract-core++

Reply:
OWASP and Trivy are configured to verify different targets:
* OWASP checks each project 
* Trivy checks only Docker image
As I remember we didn't found good approach to verify gradle multiproject via Trivy.
I think using combination of OWASP and Trivy with different goals is OK for this project.

P.S. OWASP found the vulnerabilities in unused dependency (I mean in target Docker image), for example 
High severity - CVE-2022-3509 vulnerability in pkg:maven/com.google.protobuf/protobuf-java@3.19.4
bom:4.1.0 provides the 3.21.7 version
https://github.com/th2-net/th2-bom/blob/master/build.gradle#L14